template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/impala/ImpalaVolumeConfigProvider.java

```diff
@@ -43,7 +43,8 @@
                 configs.add(config(IMPALA_SCRATCH_DIRS_PARAM,
                         buildVolumePathStringZeroVolumeHandled(hostGroupView.getVolumeCount(), "impala/scratch")));
 
-                if (minAttachedVolumeSize > MIN_ATTACHED_VOLUME_SIZE_TO_ENABLE_CACHE_IN_GB) {
+                if (minAttachedVolumeSize > MIN_ATTACHED_VOLUME_SIZE_TO_ENABLE_CACHE_IN_GB
+                        && hostGroupView.getVolumeCount() > 0) {
                     long dataCacheCapacityInBytes = VolumeUtils
```

Comment:
Workaround since template validation does not prevent attachedVolumeCount as 0 even though attachedVolumeSize > 0